src/Tests/Execute/DeviceViewProvider.test.ts

```diff
@@ -97,44 +99,65 @@ suite('DeviceViewProvider', function() {
         }
-      }
+        done();
+      });
     });
-    test('get Children under Device Node', function() {
+    test('get Children under deviceManager Node', function(done) {
       let provider = new DeviceViewProvider();
-      for (const key in provider.deviceManagerMap) {
-        if (Object.prototype.hasOwnProperty.call(provider.deviceManagerMap, key)) {
-          const element = provider.deviceManagerMap[key];
-          for (const device of element.allDevices) {
-            const label = device.name;
-            const collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+      provider.loadDeviceManager('local', function(_provider: DeviceViewProvider) {
+        for (const key in provider.deviceManagerMap) {
+          if (Object.prototype.hasOwnProperty.call(provider.deviceManagerMap, key)) {
+            const element = provider.deviceManagerMap[key];
+            const label = key;
+            const collapsibleState = vscode.TreeItemCollapsibleState.Expanded;
             const managerName = key;
-            let node = new DeviceViewNode(label, collapsibleState, NodeType.device, managerName);
+            let node =
+                new DeviceViewNode(label, collapsibleState, NodeType.deviceManager, managerName);
             let result = provider.getChildren(node);
-                    assert.strictEqual(result.length, element.findDevice(label)?.availableExecutors.size);
+            assert.strictEqual(result.length, element.allDevices.length);
           }
         }
-      }
+        done();
+      });
+    });
+    test('get Children under Device Node', function(done) {
+      let provider = new DeviceViewProvider();
+      provider.loadDeviceManager('local', function(_provider: DeviceViewProvider) {
+        for (const key in provider.deviceManagerMap) {
+          if (Object.prototype.hasOwnProperty.call(provider.deviceManagerMap, key)) {
+            const element = provider.deviceManagerMap[key];
+            for (const device of element.allDevices) {
+              const label = device.name;
+              const collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+              const managerName = key;
+              let node = new DeviceViewNode(label, collapsibleState, NodeType.device, managerName);
+              let result = provider.getChildren(node);
+              // as currently no executor registered on this test,
+              // this will return single view with 'No subtree available'.
+              assert.strictEqual(result.length, 1);
+              assert.strictEqual(result[0].label, 'No subtree available');
+              assert.strictEqual(result[0].collapsibleState, vscode.TreeItemCollapsibleState.None);
+              assert.strictEqual(result[0].nodetype, NodeType.none);
+              assert.strictEqual(result[0].managerName, '');
+            }
+          }
+        }
+        done();
+      });
+    });
+    suite('#refresh', function() {
```

Comment:
I think there is no test in this suite... was it intended?

Reply:
Oh I will check this. thanks!

Reply:
um....I think we don't need to check this. I forget to remove this. sorry for confusing.